### PR TITLE
fix: [DebugBar] scroll to top

### LINKED ISSUE #8594: Bug: Debugbar Click(toggle) Scrolls Document to Top
### PHP Version

8.2, 8.3

### CodeIgniter4 Version

4.4.6

### CodeIgniter4 Installation Method

Manual (zip or tar.gz)

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

fpm-fcgi

### Database

Mariadb

### What happened?

CI 4.4.6 : When clicking on Debug toolbar, the document is scrolled automatically to the TOP, as if someone had clicke an achor link to body tag.

Does not happen in 4.4.1, So I inspected the a href tags used (in both the CI versions) for the toggle buton of the debugtoolbar.

```
4.4.1 : <a id="debug-bar-link" href="**javascript:void(0)**" title="Open/Close">
4.4.6 : <a id="debug-bar-link" href="**#**" title="Open/Close">
```

I believe somewhere in the toolbar.js either return false is required, or preventdefault is to be used.

This can mess up with JS based routing as well I believe. 

                    
        

### Steps to Reproduce

Switch to development mode : 

1) Create a Page that has content more than the window height so as to show scrollbars on right side.
2) Scroll down to the bottom of the page.
3) Click on the debugtoolbar toggle button.

It can be noticed that the page scrolls to the top.

### Expected Output

Page position should remain as it is and should not scroll at all.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ddevsr**
**Description**
Fixed #8594

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by acpadhi** [comment]
Thanks, however, this does not fix issue across the debugbar. Only fixes the toggle icon. I have uploaded 2 files in #8594 which seem to fix it for me across all links in the debugbar. Please refer to #8594 to download a zip with the changed files.

**Comment by ddevsr** [comment]
@acpadhi Updated all button

**Comment by acpadhi** [comment]
Line Number : 31 - role="button" missing, so when debugbar is closed, cursor change not working when hover on toggle button.
`<div id="debug-icon" class="debug-bar-ndisplay">
    <a id="debug-icon-link">
`

**Comment by kenjis** [comment]
@acpadhi 

> when debugbar is closed, cursor change not working when hover on toggle button.

What do you mean? What is the toggle button?

**Comment by ddevsr** [comment]
@acpadhi Can you record?

File: admin/css/debug-toolbar/toolbar.scss [hunk 1]
```diff
@@ -36,6 +39,9 @@
     clear: both;
     text-align: center;
 
+    // Cursor
```
Comment (paulbalandan) [inline admin/css/debug-toolbar/toolbar.scss:42]: No need for this comment, in my opinion.

**Comment by acpadhi** [comment]
@kenjis 
Bottom right corner in development mode in debugbar closed state.
image
MouseOver on that icon does not change as exepected, since role=button is missing for that anchor tag.

**Comment by acpadhi** [comment]
@ddevsr 
Since you are working on the CSS/SCSS, please look into this as well : 

image

In debugbar when someone clicks on the Views button to show the views higlighted in the page, the button in the toolbar should also have a darker background to highlight that the views button has been toggled to an ON state, just like how the other buttons in the debugbar behave. If this is confusing, please let me know and I will write back again to explain.

**Comment by kenjis** [comment]
@acpadhi 

https://github.com/codeigniter4/CodeIgniter4/assets/87955/678971b1-d264-438f-ac77-fbb7cf2999db

**Comment by kenjis** [comment]
> In debugbar when someone clicks on the Views button to show the views higlighted in the page, the button in the toolbar should also have a darker background to highlight that the views button has been toggled to an ON state, just like how the other buttons in the debugbar behave. 

"Views" tab shows nothing, so isn't the behavior intentional?
At least, the behavior is the same in v4.0.1.

Clicked "Files":
Screenshot 2024-03-05 14 38 17

and clicked "Views":
Screenshot 2024-03-05 14 38 27

**Comment by acpadhi** [comment]
> "Views" tab shows nothing, so isn't the behavior intentional?
> At least, the behavior is the same in v4.0.1.

In my Opinion, it is kind of not directly imminent that that the view button is clicked, I understant that there is border highlithing on page view areas, but when working on a large pages, visually it can be distracting to see gaps in design and wondering if some css is causing it. A dark background in the button will immediately convey "Oh my debugbar view button is toggled, so I see the gaps" . This is not a pressing thing, just a suggestion to keep consistency, since afterall it is a  toggle, irrespective of whenther showing additional content or not.

**Comment by kenjis** [comment]
@acpadhi Ah, I got what you say. You say it is better that when the "Views" is on, the "Views" is darker. 
It would certainly be better that way.

**Comment by acpadhi** [comment]
> Ah, I got what you say. You say it is better that when the "Views" is on, the "Views" is darker.
> It would certainly be better that way.

Yes, that way this Views button will be conistent with other buttons on the visual aspect, and also make it exlicit that it is in ON state.

**Comment by kenjis** [comment]
@acpadhi I agree with your opinion. But the current behavior is not a bug. So changing it will be an enhancement.

This PR seems good to me.

**Review by kenjis — APPROVED** [review]

**Comment by ddevsr** [comment]
@acpadhi Ah, i see. If view button already click, marking is ACTIVE right?

@kenjis Next PR?

**Comment by kenjis** [comment]
@ddevsr It is an enhancement, not a bug fix. Please send a new PR to `4.5` if you send.

File: admin/css/debug-toolbar/toolbar.scss [hunk 2]
```diff
@@ -13,6 +13,9 @@
 @import '_mixins';
 @import '_settings';
 
+a[role="button"] {
+    cursor: pointer;
+}
```
Comment (kenjis) [inline admin/css/debug-toolbar/toolbar.scss:18]: Could this change break existing pages?

Comment (ddevsr) [reply admin/css/debug-toolbar/toolbar.scss:18]: OK, I change to new css

File: admin/css/debug-toolbar/toolbar.scss (hunk 1 above)
Comment (ddevsr) [reply admin/css/debug-toolbar/toolbar.scss]: Done

**Comment by kenjis** [comment]
The following error will be fixed by #8612.
```
 Error: Ignored error pattern #^Cannot unset offset string on array\<int, int\>\|null\.$# in path /home/runner/work/CodeIgniter4/CodeIgniter4/system/Test/Mock/MockCache.php was not matched in reported errors.
 ------ ---------------------------------------------------------------------------- 
  Line   system/Test/Mock/MockCache.php                                              
 ------ ---------------------------------------------------------------------------- 
         Ignored error pattern #^Cannot unset offset string on array<int,            
         int>\|null\.$# in path                                                      
         /home/runner/work/CodeIgniter4/CodeIgniter4/system/Test/Mock/MockCache.php  
         was not matched in reported errors.                                         
 ------ ---------------------------------------------------------------------------- 
```
https://github.com/codeigniter4/CodeIgniter4/actions/runs/8210351538/job/22457631537?pr=8595

**Review by kenjis — APPROVED** [review]
LGTM!

**Review by paulbalandan — APPROVED** [review]

**Pull request merged**